doc/source/gr_generic.rst

```diff
@@ -220,6 +224,12 @@ To do: move to ``gr_vec``
 
 .. function:: int gr_generic_vec_is_zero(gr_srcptr vec, slong len, gr_ctx_t ctx)
 
+.. function:: int gr_generic_vec_contains_zero_vec(gr_srcptr vec, slong len, gr_ctx_t ctx)
+
+    Returns whether *vec* contains the zero vector, in the sense that all
+    elements contain the zero element of the base ring, as described
+    in :func:`gr_contains_zero`.
+
```

Comment:
Perhaps just stick to the current semantic (that is, `gr_generic_vec_contains_zero`) since the vector itself is an element in a commutative group?

Reply:
The same follows for all other structures, I suppose.

Reply:
I went with the other name(s) because `_contains_zero` for vectors, matrices and polynomials could be more easily misinterpreted as "contains at least one (exact) zero, somewhere". With the current naming you avoid that particular connotation. You could still think it might mean "contains a (exact) zero sub-structure (e.g. matrix, vector)", but then you notice there is no argument indicating the size of a hypothetical sub-structure, and if it was about an arbitrary sub-structure then that would include a 1x1 sub-structure (a single zero), so then mentioning a sub-structure would be redundant, so it's not that.

Maybe a third person can weigh in with their opinion on the naming. I can change the names if needed.

Reply:
But we do not name addition of two vectors `_gr_vec_add_vec`, instead we name it `_gr_vec_add`. And for scalars, it is named `_gr_vec_add_scalar`.

What about `contains_zero` and `contains_zero_entry`?

I would argue that we otherwise remove the mathematical aspect from the module since we no longer consider the choice of basis.